### PR TITLE
Add more examples to about_Operators

### PR DESCRIPTION
**Pull request by yecril71pl**
# PR Summary

I find it frustrating to see a cast and having to evaluate its value, so I decided to provide explicit results.
Also, fun with hexadecimal strings (&incare; [BaseNumberConverter](https://docs.microsoft.com/en-us/dotnet/api/system.componentmodel.basenumberconverter.convertfrom)).

## PR Context

Select the area of the Table of Contents containing the documents being changed.

**Conceptual content**
- [ ] Overview and Install
- [ ] Learning PowerShell
  - [ ] PowerShell 101
  - [ ] Deep dives
  - [ ] Sample scripts
  - [ ] Remoting
- [ ] Release notes (What's New)
- [ ] Windows PowerShell
  - WMF, ISE, release notes, etc.
- [ ] DSC articles
- [ ] Community resources
- [ ] Gallery articles
- [ ] Scripting and development
  - [ ] Language Spec
  - [ ] Legacy SDK

**Cmdlet reference & about_ topics**
- [X] Preview content
- [X] Version 7.1 content
- [X] Version 7.0 content
- [X] Version 5.1 content

## PR Checklist

- [X] I have read the [contributors guide][contrib] and followed the style and process guidelines
- [X] PR has a meaningful title
- [X] PR is targeted at the _staging_ branch
- [X] All relevant versions updated
- [ ] Includes content related to issues and PRs - see [Closing issues using keywords][key].
- [X] This PR is ready to merge and is not **Work in Progress**
  - If the PR is work in progress, please add the prefix `WIP:` or `[WIP]` to the beginning of the
    title and remove the prefix when the PR is ready.

[contrib]: https://docs.microsoft.com/powershell/scripting/community/contributing/overview
[key]: https://help.github.com/en/articles/closing-issues-using-keywords

### CODE REVIEW

File: reference/7.1/Microsoft.PowerShell.Core/About/about_Operators.md [hunk 1]
```diff
@@ -337,10 +337,18 @@ Converts or limits objects to the specified type. If the objects cannot be
 converted, PowerShell generates an error.
 
 ```powershell
-[DateTime]"2/20/88" - [DateTime]"1/20/88"
-[Int] (7/2)
-[String] 1 + 0
-[Int] '1' + 0
+# Dates
+[DateTime] '2/20/88' - [DateTime] '1/20/88' -EQ [TimeSpan] '31'
```
Comment (sdwheeler) [inline reference/7.1/Microsoft.PowerShell.Core/About/about_Operators.md:341]: This is a good replacement for the date example.

File: reference/7.1/Microsoft.PowerShell.Core/About/about_Operators.md [hunk 2]
```diff
@@ -337,10 +337,18 @@ Converts or limits objects to the specified type. If the objects cannot be
...
+# Bankers’ rounding
+[Int] (7/2) -EQ 4 -AND [Int] (9/2) -EQ 4
+# Concatenation
+[String] 1 + 0 -EQ '10'
+# Addition
+[Int] '1' + 0 -EQ 1
```
Comment (sdwheeler) [inline reference/7.1/Microsoft.PowerShell.Core/About/about_Operators.md:347]: I don't think we need these examples. They are just more of the same information. This is also covered in about_Arithmetic_Operators, which is a better place for that discussion.

File: reference/7.1/Microsoft.PowerShell.Core/About/about_Operators.md [hunk 3]
```diff
@@ -337,10 +337,18 @@ Converts or limits objects to the specified type. If the objects cannot be
...
+# Hexadecimal
+[Int] '0XF' -EQ 0XF -AND [Int] '#F' -EQ 0XF -AND [Int] '&HF' -EQ 0XF
+# Binary
+[Int] '0B1111' -EQ 0B1111
```
Comment (sdwheeler) [inline reference/7.1/Microsoft.PowerShell.Core/About/about_Operators.md:351]: Hex and binary numbers are covered, in detail, in about_Numeric_Literals.

**Review by sdwheeler — CHANGES_REQUESTED** [review]
See my comments below and the comments in #8281

**Comment by sdwheeler** [comment]
Thanks for the submission. Also, see my comments in #8282.

Also, when submitting changes to a file, the changes need to be made to all versions of the file. You have submitted two PRs for the same file (about_Operators). Please combine the changes in one PR.

If you want to wait for feedback before copying the changes to the other versions, mark the PR as a Work-in-Progress [WIP] as described in the PR template. After the feedback you can make the necessary changes to all version and update the PR.

File: reference/7.1/Microsoft.PowerShell.Core/About/about_Operators.md (hunk 3 above)
Comment (yecril71pl) [reply reference/7.1/Microsoft.PowerShell.Core/About/about_Operators.md:351]: Observe that `#F` is not a numeric literal but `[INT] '#F'` is valid.

Comment (sdwheeler) [reply reference/7.1/Microsoft.PowerShell.Core/About/about_Operators.md]: Yes, but this documentation is not about all the different ways the type-casting can convert values. This is about the existence of the `[type]` cast operator. These more explicit examples belong elsewhere.

Comment (yecril71pl) [reply reference/7.1/Microsoft.PowerShell.Core/About/about_Operators.md]: I gave these specific examples as a hint that conversions are weird (sometimes) and the coder should be prepared to handle unexpected results.  I have removed the examples at your request.

**Review by sdwheeler — CHANGES_REQUESTED** [review]
This change looks good. Please copy the changes to the other versions.

**Pull request merged**